packages/@ember/debug/lib/deprecate.ts

```diff
@@ -204,32 +202,8 @@ if (DEBUG) {
     assert(missingOptionsDeprecation, Boolean(options && (options.id || options.until)));
     assert(missingOptionsIdDeprecation, Boolean(options!.id));
     assert(missingOptionsUntilDeprecation, Boolean(options!.until));
-
-    if (!options!.for && !FOR_MISSING_DEPRECATIONS.has(options!.id)) {
-      FOR_MISSING_DEPRECATIONS.add(options!.id);
-
-      deprecate(missingOptionsForDeprecation(options!.id), Boolean(options!.for), {
-        id: 'ember-source.deprecation-without-for',
-        until: '4.0.0',
-        for: 'ember-source',
-        since: {
-          enabled: '3.24.0',
-        },
-      });
-    }
-
-    if (!options!.since && !SINCE_MISSING_DEPRECATIONS.has(options!.id)) {
-      SINCE_MISSING_DEPRECATIONS.add(options!.id);
-
-      deprecate(missingOptionsSinceDeprecation(options!.id), Boolean(options!.since), {
-        id: 'ember-source.deprecation-without-since',
-        until: '4.0.0',
-        for: 'ember-source',
-        since: {
-          enabled: '3.24.0',
-        },
-      });
-    }
+    assert(missingOptionsForDeprecation(options!.id), Boolean(options!.for));
+    assert(missingOptionsSinceDeprecation(options!.id), Boolean(options!.since));
```

Comment:
Seems inconsistent that these asserts bother to include the deprecation id and the earlier asserts do not.

Reply:
Yeah I wasn't sure why, but seems like good so I left it